libpromises/eval_context.c

```diff
@@ -192,8 +192,21 @@ struct EvalContext_
     RemoteVarPromisesMap *remote_var_promises;
 
     bool dump_reports;
+    bool override_immutable;
```

Comment:
I still don't like this being "so global", it's a directive for one specific promise based on its attributes. We cannot rely solely on the attributes, fine, but then this should be in the `Promise` struct or the promise's stack frame.